frontend/src/context/SocketContext.tsx

```diff
@@ -0,0 +1,70 @@
+import { PropsWithChildren, createContext, useEffect, useState } from "react";
+import { socket } from "./socket";
+import { InitialSocketContext } from "../types/InitialSocketContext";
+
+const intialContext: InitialSocketContext = {
```

Comment:
So a better and more common practice way to do this nowadays is to set the initial state to null and then have a hook that checks for the presence of a context. I can show you soon. It would eliminate a lot of this code